### PR TITLE
Add Green model to benchmark 

### PR DESCRIPTION
**Pull request by jpaillard**
As discussed, I recently tested the model that we proposed on the benchopt benchmark.
 - paper: https://www.biorxiv.org/content/10.1101/2024.05.14.594142v2.abstract 
 - code: https://github.com/Roche/neuro-green

I thought it could be interesting to contribute to the public repo by adding it as a new solver.

### CODE REVIEW

**Review by bruAristimunha — APPROVED** [review]

**Comment by bruAristimunha** [comment]
When everything is green @jpaillard, it's ok for me to apply the merge!

File: solvers/green.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            pool_layer=RealCovariance(),
+            bi_out=self.bi_out,
+            dtype=torch.float32,
+            out_dim=n_classes,
+        )
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        if torch.cuda.is_available():
+            torch.backends.cudnn.benchmark = True
+
+        seed = 0
```
Comment (tomMoral) [inline solvers/green.py:69]: This should be a parameter of the class so we can change it in the repetitions?

File: solvers/green.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            classes=list(range(n_classes)),
+            callbacks=callbacks,
+        )
+
+        self.clf = clf
+
+        self.X = X
+        self.y = y
+
+    def run(self, n_iter):
```
Comment (tomMoral) [inline solvers/green.py:103]:
```suggestion
    def run(self, _):
```
So it is less confusing.

File: solvers/green.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        # here we want to define a function that gets the data X,y from Moabb
+        # and converts it to data accessible for deep learning methods
+        lr = self.lr
+        n_classes = len(set(y))
+
+        n_channels = X[0].shape[0]
+
+        model = get_green(
+            n_freqs=self.n_freqs,
+            kernel_width_s=0.5,
```
Comment (tomMoral) [inline solvers/green.py:53]: Why no make this a parameter?

**Review by tomMoral — COMMENTED** [review]
Looks good! Thanks @jpaillard 
Here are a few changes that would make the solver better fit `benchopt`.

**Comment by tomMoral** [comment]
Open a new PR.

t

**Pull request merged**